modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,371 @@
+import { afterEach, beforeEach, context, describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import * as Chai from 'chai';
+import { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+import type Editor from 'tinymce/core/api/Editor';
+import { createUserLookup, type User } from 'tinymce/core/lookup/UserLookup';
+
+Chai.use(chaiAsPromised);
```

Comment:
Does introducing this dependency give us enough gain?
To me couldn't this all be just `await` and regular asserts? The rejected ones can be wrapped in try/catch of a catch that resolves to the error value and awaited. I guess it's a bit more explicit on what is going on.

Reply:
We started with that, but Kim has quite a few tests for rejected cases and it turns into a lot of hand-written code noise with try/catch + the `assert.fail()` if it doesn't throw.

Chai was already a dependency, and it has a "throws error" assertion, so I think it's reasonable to add a promise-based "is rejected" assertion. I am a little surprised it isn't built-in.

Reply:
I see now that it's maintained by the same team. Then I have less issues with it. I'm always worried about just pulling in dependencies that gets us to dependency hell.